### ISSUE TITLE
TypeError: unsupported operand type(s) for +: 'int' and 'str' | wmt_ende_tokens_32k

### ISSUE THREAD

**Issue by mehmedes** [tensorflow/tensor2tensor#81]
When running ``wmt_ende_tokens_32k`` on new corpus data, I encounter the following error. Could you please help me with this?
```
INFO:tensorflow:Generating training data for wmt_ende_tokens_32k.
INFO:tensorflow:Not downloading, file already found: /tmp/t2t_datagen/corpus.tgz
INFO:tensorflow:Reading file: corpus.en
INFO:tensorflow:Reading file: corpus.de
INFO:tensorflow:Alphabet contains 156 characters
Traceback (most recent call last):
  File "/usr/local/bin/t2t-datagen", line 378, in <module>
    tf.app.run()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/platform/app.py", line 48, in run
    _sys.exit(main(_sys.argv[:1] + flags_passthrough))
  File "/usr/local/bin/t2t-datagen", line 361, in main
    training_gen(), FLAGS.problem + UNSHUFFLED_SUFFIX + "-train",
  File "/usr/local/bin/t2t-datagen", line 151, in <lambda>
    lambda: wmt.ende_wordpiece_token_generator(FLAGS.tmp_dir, True, 2**15),
  File "/usr/local/lib/python2.7/dist-packages/tensor2tensor/data_generators/wmt.py", line 188, in ende_wordpiece_token_generator
    tmp_dir, "tokens.vocab.%d" % vocab_size, vocab_size)
  File "/usr/local/lib/python2.7/dist-packages/tensor2tensor/data_generators/generator_utils.py", line 223, in get_or_generate_vocab
    vocab_size, tokenizer.token_counts, vocab_filepath, 1, 1e3)
  File "/usr/local/lib/python2.7/dist-packages/tensor2tensor/data_generators/text_encoder.py", line 329, in build_to_target_size
    return bisect(min_val, max_val)
  File "/usr/local/lib/python2.7/dist-packages/tensor2tensor/data_generators/text_encoder.py", line 313, in bisect
    present_count = (max_val + min_val) // 2
TypeError: unsupported operand type(s) for +: 'int' and 'str'
```

**Comment by lukaszkaiser** [comment]
Should be corrected on head in 1.0.10 (couldn't push to pypi yet, but coming...)

**Comment by mehmedes** [comment]
@lukaszkaiser  I've just tried commit ``a013fce``, which results in the following:
```INFO:tensorflow:Generating training data for wmt_ende_tokens_32k.
INFO:tensorflow:Not downloading, file already found: /tmp/t2t_datagen/corpus.tgz
INFO:tensorflow:Reading file: corpus.en
/usr/local/lib/python2.7/dist-packages/tensor2tensor/data_generators/tokenizer.py:75: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  is_alnum = [c in self._ALPHANUMERIC_CHAR_SET for c in text]
/usr/local/lib/python2.7/dist-packages/tensor2tensor/data_generators/tokenizer.py:79: UnicodeWarning: Unicode unequal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  if token != u" " or token_start == 0:
INFO:tensorflow:Reading file: corpus.de
Traceback (most recent call last):
  File "/usr/local/bin/t2t-datagen", line 378, in <module>
    tf.app.run()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/platform/app.py", line 48, in run
    _sys.exit(main(_sys.argv[:1] + flags_passthrough))
  File "/usr/local/bin/t2t-datagen", line 361, in main
    training_gen(), FLAGS.problem + UNSHUFFLED_SUFFIX + "-train",
  File "/usr/local/bin/t2t-datagen", line 151, in <lambda>
    lambda: wmt.ende_wordpiece_token_generator(FLAGS.tmp_dir, True, 2**15),
  File "/usr/local/lib/python2.7/dist-packages/tensor2tensor/data_generators/wmt.py", line 188, in ende_wordpiece_token_generator
    tmp_dir, "tokens.vocab.%d" % vocab_size, vocab_size)
  File "/usr/local/lib/python2.7/dist-packages/tensor2tensor/data_generators/generator_utils.py", line 223, in get_or_generate_vocab
    vocab_size, tokenizer.token_counts, vocab_filepath, 1, 1e3)
  File "/usr/local/lib/python2.7/dist-packages/tensor2tensor/data_generators/text_encoder.py", line 318, in build_to_target_size
    return bisect(min_val, max_val)
  File "/usr/local/lib/python2.7/dist-packages/tensor2tensor/data_generators/text_encoder.py", line 300, in bisect
    present_count = (max_val + min_val) // 2
TypeError: unsupported operand type(s) for +: 'int' and 'str'
```

**Comment by mehmedes** [comment]
I hadn't updated ``wmt.py`` and ``generator_utils.py``. 
After updating and rerunning ``a013fce``, t2t says:
```
...
UnicodeDecodeError: 'ascii' codec can't decode byte 0xa4 in position 0: ordinal not in range(128)
```

**Comment by mehmedes** [comment]
#74

**Comment by lukaszkaiser** [comment]
This is hopefully corrected in 1.0.11. please give it a try. I'm closing for now, please reopen if you still see the issue.

**Comment by mehmedes** [comment]
Thank you! Works perfectly.
